src/asset.ts

```diff
@@ -11,28 +11,52 @@ class Asset {
   public ledger: Ledger;
   public name: string;
   public contract?: string;
+  public decimals?: number;
 
-  public constructor(name: string, ledger: Ledger, contract?: string) {
+  public constructor(
```

Comment:
You could add an abstraction to `Asset` and have `NativeAsset` and `Erc20Asset`? This would simplify the constructor and the odd naming of `toNominalUnitForToken`.

Reply:
From a functional point of view I am not sure there is much gain to an added abstraction, I'll consider.